src/compiler/commandLineParser.ts

```diff
@@ -584,4 +581,233 @@ namespace ts {
 
         return { options, errors };
     }
+
+    const invalidTrailingRecursionPattern = /(^|\/)\*\*\/?$/;

```

Comment:
Can you add comments to all these regular expressions.. they are impossible to parse.